paddle/fluid/framework/ir/CMakeLists.txt

```diff
@@ -41,6 +41,9 @@ pass_library(seq_concat_fc_fuse_pass inference)
 pass_library(conv_bn_fuse_pass inference)
 
 cc_library(fuse_elewise_add_act_pass SRCS fuse_elewise_add_act_pass.cc DEPS pass graph_pattern_detector )
+if(WITH_MKLDNN)
+  pass_library(conv_elementwise_add_mkldnn_fuse_pass inference)
+endif()
```

Comment:
line 32, 45 can refine:
```
if (WITH_MKLDNN)
   pass_library(conv_relu_mkldnn_fuse_pass inference)
   pass_library(conv_elementwise_add_mkldnn_fuse_pass inference)
endif()
```

Reply:
Is the error still present when #13813 is turned on?

Reply:
True, now I saw it. I will start looking at it.

Is test_analyzer_resnet50 part of your CI testing suite?

Reply:
Yes, it is. But `config._use_mkldnn=false` in `test_analyzer_resnet50` now.

Reply:
Ok, understood. Thanks!

Reply:
@luotao1 I checked the code that you showed in your diff, and for me residual connection pass (`conv_elementiwse_add_mkldnn_fuse_pass`) does not detect any patterns, and the test passes. There are some biases implemented with `elementwise_add` that should be fused first.

I think the correct order of MKLDNN passes should be:
```
conv_bias_mkldnn_fuse_pass
conv_relu_mkldnn_fuse_pass
conv_elementwise_add_mkldnn_fuse_pass
```

Reply:
Thanks for pointing that out. I will check it.

Reply:
@luotao1 I checked it on the branch where I merged also #13858. I haven't encountered the problem with `Bias` input in MKLDNN convolution. I actually haven't seen residual connection fusion pass being executed in `test_analyzer_ocr` test.

However, I found another issue with correct order of parameters in `elementwise_add` operator in `test_analyzer_resnet50`. I submitted a new commit that solves this issue. Could you try it now?

If it makes it easier for you, I prepared a branch where I tested issues you mentioned. It a branch with conv+elementwise_add fusion and conv+bias fusion (#13858). I can share it with you.

Reply:
Thanks for your fix. I still have some questions:
- Must this PR be tested and merged after #13858? 
- In your latest https://github.com/PaddlePaddle/Paddle/pull/13485/commits/e7060e9073d53bd38ddb8c150d083c360db94f7a, I could not find `conv_elementwise_add_mkldnn_fuse_pass` in `paddle/fluid/inference/analysis/analyzer.h`. Could you add this pass in `analyzer.h`?
- How about test this PR with #13909?

Reply:
I think #13858 should be merged first, then this PR, and then the PR that turns on the MKLDNN passes.

I also added using `conv_elementwise_add_mkldnn_fuse_pass` in analyzer.h, and an additional fix for inference transpiler.

I will check MKLDNN fusion passes with #13909.

Reply:
@luotao1 I created a branch with #13858, #13909 and this PR. `test_analyzer_resnet50` and `test_analyzer_ocr` work but I had to add the following pieces of code:

```
diff --git a/paddle/fluid/inference/tests/api/analyzer_resnet50_tester.cc b/paddle/fluid/inference/tests/api/analyzer_resnet50_tester.cc
index 92cc76d..72b52f0 100644
--- a/paddle/fluid/inference/tests/api/analyzer_resnet50_tester.cc
+++ b/paddle/fluid/inference/tests/api/analyzer_resnet50_tester.cc
@@ -98,6 +98,9 @@ TEST(Analyzer_resnet50, compare) {
   // with native_outputs as well.
   FLAGS_use_MKLDNN = false;
   AnalysisConfig cfg1;
+  cfg1.ir_passes.push_back("conv_bias_mkldnn_fuse_pass");
+  cfg1.ir_passes.push_back("conv_relu_mkldnn_fuse_pass");
+  cfg1.ir_passes.push_back("conv_elementwise_add_mkldnn_fuse_pass");
   SetConfig(&cfg1);
   CompareNativeAndAnalysis(cfg1, input_slots_all);
   FLAGS_use_MKLDNN = true;
diff --git a/paddle/fluid/inference/tests/api/analyzer_vis_tester.cc b/paddle/fluid/inference/tests/api/analyzer_vis_tester.cc
index 96a3c6f..6ad6d6d 100644
--- a/paddle/fluid/inference/tests/api/analyzer_vis_tester.cc
+++ b/paddle/fluid/inference/tests/api/analyzer_vis_tester.cc
@@ -131,6 +131,9 @@ TEST(Analyzer_vis, compare) {
   // with native_outputs as well.
   FLAGS_use_MKLDNN = false;
   AnalysisConfig cfg1;
+  cfg1.ir_passes.push_back("conv_bias_mkldnn_fuse_pass");
+  cfg1.ir_passes.push_back("conv_relu_mkldnn_fuse_pass");
+  cfg1.ir_passes.push_back("conv_elementwise_add_mkldnn_fuse_pass");
   SetConfig(&cfg1);
   CompareNativeAndAnalysis(cfg1, input_slots_all);
   FLAGS_use_MKLDNN = true;
```

Reply:
I see it.

Reply:
@luotao1 My build failed with the following message:
```
[Step 1/1] error: [Errno 28] No space left on device
```
Do you know anything about it?

Reply:
@tpatejko Since #13958 and #13909 are merged, is this PR still be merged after #13858?

Reply:
Yes.

Reply:
@tpatejko #13858 is merged.